instrumentation/vertx-web-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/RoutingContextHandlerWrapper.java

```diff
@@ -36,6 +40,24 @@ public void handle(RoutingContext context) {
     } catch (Exception ex) {
       log.error("Failed to update server span name with vert.x route", ex);
     }
-    handler.handle(context);
+    try {
+      handler.handle(context);
+    } catch (Throwable throwable) {
+      if (serverSpan != null && serverSpan.getSpanContext().isValid()) {
```

Comment:
if it's not valid, I think span will be no-op anyways
```suggestion
      if (serverSpan != null) {
```